src/client/runner/automation/get-element.js

```diff
@@ -0,0 +1,73 @@
+import hammerhead from '../deps/hammerhead';
+import testCafeCore from '../deps/testcafe-core';
+import cursor from './cursor';
+
+var browserUtils  = hammerhead.utils.browser;
+var positionUtils = testCafeCore.positionUtils;
+var domUtils      = testCafeCore.domUtils;
+
+
+function getElementFromPoint (x, y) {
+    var cursorVisible = cursor.visible;
+
+    if (cursorVisible)
+        cursor.hide();
+
+    var topElement = positionUtils.getElementFromPoint(x, y);
+
+    if (cursorVisible)
+        cursor.show();
+
+    return topElement;
+}
+
+function ensureImageMap (imgElement, areaElement) {
+    var mapElement = domUtils.closest(areaElement, 'map');
+
+    return mapElement && mapElement.name === imgElement.useMap.substring(1) ? areaElement : imgElement;
+}
+
+function findElementOrNonEmptyChildFromPoint (x, y, element) {
+    var topElement = positionUtils.getElementFromPoint(x, y);
+
+    if (topElement && topElement === element || domUtils.containsElement(element, topElement) && topElement.textContent.length)
```

Comment:
This one it too long. Consider splitting it into boolean vars with meaningful names